### DISCUSSION TITLE
Is it possible to control how much space widgets take without manually doing out the layout for everything?

### DISCUSSION THREAD

**Discussion by n00kii** [emilk/egui#1754] [Q&A] [answered]
Say I wanted a spinner that's 100px by 100px. I know you can do `Spinner::new().size()` to control how big it is in the sense of scaling up the whole thing, but what if I wanted to put a normal sized spinner in a frame that's 100px by 100p? Or even just a normal label into a frame of a certain specific size, where it wraps if the text isn't long enough. I found `egui::containers::Frame` but that doesn't seem to be what I'm looking for, you can set a margin but not a specific max size of the contents of the inner widget. Is there any way currently to do what I'm describing?

**Comment by dacid44** [comment, marked as answer]
Have you tried `Ui::add_sized()`? https://docs.rs/egui/latest/egui/struct.Ui.html#method.add_sized
